Assets/MirageXR/DataModel/DataModel.cs

```diff
@@ -0,0 +1,319 @@
+        [JsonProperty] public Location Location { get; set; }
+        [JsonProperty] public List<Trigger>? Triggers { get; set; } // cannot be null?
+        [JsonProperty] public List<Comment> Comments { get; set; }
+        [JsonProperty] public List<RequiredToolsPartsMaterials> RequiredToolsPartsMaterials { get; set; }
+        [JsonProperty] public List<File>? Attachment { get; set; }
+        [JsonProperty] public List<Comment> PrivateNotes { get; set; }
+    }
+
+    [Serializable]
+    public class Comment : DataModel
+    {
+        [JsonProperty] public User User { get; set; }
+        [JsonProperty] public string Text { get; set; }
+        [JsonProperty] public List<File>? Attachment { get; set; }
+    }
+
+
+    [Serializable]
+    public class RequiredToolsPartsMaterials
+    {
+        [JsonProperty] public string ToolPartMaterial { get; set; }
+    }
+
+    [Serializable]
+    public class User : DataModel
+    {
+        [JsonProperty] public string Name { get; set; }
+    }
+
+    [Serializable]
+    public class Content : DataModel
+    {
+        [JsonProperty] public ContentType Type { get; set; }
+        [JsonProperty] public string Name { get; set; }
+        [JsonProperty] public string Description { get; set; }
+        [JsonProperty] public bool IsVisible { get; set; }
+        [JsonProperty] public Location Location { get; set; }
+        [JsonProperty] public JToken ContentData { get; set; }
+
+        public Content<T> ToSpecifiedContent<T>() where T : ContentData
+        {
+            return new Content<T>
+            {
+                Id = Id,
+                Name = Name,
+                Version = Version,
+                Type = Type,
+                ContentData = ContentData.ToObject<T>(),
+                Location = Location
+            };
+        }
+    }
+
+    [Serializable]
+    public class File : DataModel
+    {
+        [JsonProperty] public string Name { get; set; }
+        [JsonProperty] public string FileHash { get; set; }
+    }
+
+    public enum ContentType
+    {
+        Unknown,
+        Image,
+        Video,
+        Audio,
+        Ghost,
+        Label,
+        Act,
+        Effects,
+        Model,
+        Character,
+        Interaction, //pick and place
+        ImageMarker,
+        Plugin,
+        Drawing,
+        Instructor //virtual instructor
+    }
+
+    [Serializable]
+    public class Content<T> : DataModel where T : ContentData
+    {
+        [JsonProperty] public ContentType Type { get; set; }
+        [JsonProperty] public string Name { get; set; }
+        [JsonProperty] public Location Location { get; set; }
+        [JsonProperty] public T ContentData { get; set; }
+    }
+
+    [Serializable]
+    public class Trigger    //todo
+    {
+        [JsonProperty] public TriggerType Type { get; set; }
+        [JsonProperty] public Guid StepId { get; set; } // step.id
+        [JsonProperty] public string Event { get; set; } // string needs to be eval’d
```

Comment:
Maybe we need to separate between:
* `ListenToEvent` (e.g. "onIncorrectPlacement" from pick and place: this trigger listens to this event, and if it occurs, and if it is an Event of type trigger, then it triggers the trigger) 
* and `SendEventOnTrigger` (an event to be thrown, when the trigger conditions are fulfilled - to inform all augmentations about trigger, e.g. "show"/"hide"/"animate")

Reply:
`OutputEventParameter` + `OutputEventParameterValue`

Reply:
* Mikhail and I think that the trigger should also have a `ContentId` so that it is linked with its augmentation